utils/dataset_validator.py

```diff
@@ -0,0 +1,116 @@
+from flask_uuid import UUID_RE
+
+DATASET_NAME_LEN_MIN = 1
+DATASET_NAME_LEN_MAX = 100
+
+CLASS_NAME_LEN_MIN = 1
+CLASS_NAME_LEN_MAX = 100
+
+
+def validate(dataset):
+    """Validator for datasets.
+
+    Dataset must have the following structure:
+    {
+        - name (string)
+        - description (string, optional)
+        - classes (list of class dicts)
+            {
+                - name (string)
+                - description (string, optional)
+                - recordings (list of UUIDs)
+            }
+    }
+
+    Complete dataset must contain at least two classes with two recordings in
+    each class.
+
+    Args:
+        dataset: Dataset stored in a dictionary.
+
+    Raises:
+        ValidationException: A general exception for validation errors.
+        IncompleteDatasetException: Raised in cases when one of "completeness"
+            requirements is not satisfied.
+    """
+    if not isinstance(dataset, dict):
+        raise ValidationException("Dataset must be a dictionary.")
+    _check_dict_structure(dataset, ["name", "description", "classes", "public"])
+
+    # Name
+    if "name" not in dataset:
```

Comment:
can we be more generic here - reusing the stuff you have in `_check_dict_structure` where you check for unexpected items. Just raise in there also if the expected items are not there.